### ISSUE TITLE
x2goclient: fails to find sftp binary

### ISSUE THREAD

**Issue by mkg20001** [NixOS/nixpkgs#78907]
**Describe the bug**
The x2goclient client fails to find the SFTP binary

```
x2go-ERROR-31> Unable to find the sftp-server binary. Neither bundled, nor found in $PATH nor additional directories.
```

**To Reproduce**
1. Connect to an x2go remote that has LXDE
2. Launch an LXDE session

**Expected behavior**
It should just work and show the desktop

**Screenshots**

**Additional context**

**Metadata**
 - system: `"x86_64-linux"`
 - host os: `Linux 4.19.98, NixOS, 20.03pre-git (Markhor)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.2`
 - channels(maciej): `"meros, nixpkgs-20.03pre210116.ec8fd9cc345"`
 - channels(root): `"nixos-20.03pre210639.e59dcf8ccec"`
 - nixpkgs: `/home/maciej/.nix-defexpr/channels/nixpkgs`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute:
 - x2goclient
# a list of nixos modules affected by the problem
module:
```

**Comment by bjornfor** [comment]
Ping @averelld.

**Comment by averelld** [comment]
Probably the binary only needs to be added to the search path like this: https://github.com/averelld/nixpkgs/commit/ac2b1e9915c3176879361fb5fbeab50b177fcce3 but since this feature apparently never worked, there might be more broken or maybe the recent https://bugs.x2go.org/cgi-bin/bugreport.cgi?bug=1417 actually changed something.

I will test this and make a proper PR later.

Until then, as a possible workaround should the above not work, I think it should only happen while folder-sharing / printer support is enabled (in the client session settings).

**Comment by wirew0rm** [comment]
I can confirm that disabling printer and folder sharing fixes the issue.
Let me know if you need to someone to test your fixes or otherwise help with fixing this.

**Comment by alexeymuranov** [comment]
Still an issue on NixOS 20.09 Alpha.

**Comment by alexeymuranov** [comment]
This is not stale, please remove the label.

**Comment by alexeymuranov** [comment]
I am still getting the error on NixOS 20.09:
![Screenshot from 2021-03-26 14-14-49](https://user-images.githubusercontent.com/712698/112636911-ed47ff80-8e3d-11eb-8ed0-68d217930728.png)

However, the window with a remote desktop opens and seems to work fine nonetheless.

**Comment by mkg20001** [comment]
This should be fixed in 21.11

**Comment by alexeymuranov** [comment]
@mkg20001, thanks, would it be hard to backport it to 21.05?
